WordPress/Classes/ViewRelated/Post/Scheduling/SchedulingCalendarViewController.swift

```diff
@@ -0,0 +1,199 @@
+
+        let closeButton = UIBarButtonItem(image: Gridicon.iconOfType(.cross), style: .plain, target: self, action: #selector(closeButtonPressed))
+
+        let publishButton = UIBarButtonItem(title: NSLocalizedString("Publish immediately", comment: "Immediately publish button title"), style: .plain, target: self, action: #selector(publishImmediately))
+        navigationItem.setLeftBarButton(publishButton, animated: false)
+
+        if traitCollection.verticalSizeClass == .compact {
+            navigationItem.leftBarButtonItems = [closeButton, publishButton]
+        } else {
+            navigationItem.leftBarButtonItems = [publishButton]
+        }
+    }
+
+    @objc func closeButtonPressed() {
+        dismiss(animated: true, completion: nil)
+    }
+
+    @objc func publishImmediately() {
+        coordinator?.updated(nil)
+        navigationController?.dismiss(animated: true, completion: nil)
+    }
+
+    @objc func nextButtonPressed() {
+        let vc = TimePickerViewController()
+        vc.coordinator = coordinator
+        navigationController?.pushViewController(vc, animated: true)
+    }
+}
+
+class TimePickerViewController: DatePickerSheet, DateCoordinatorHandler {
+
+    @IBOutlet weak var datePicker: UIDatePicker!
+
+    override func configureView() -> UIView {
+        let datePicker = UIDatePicker()
+        datePicker.datePickerMode = .time
+        datePicker.addTarget(self, action: #selector(timePickerChanged(_:)), for: .valueChanged)
+        if let date = coordinator?.date {
+            datePicker.date = date
+        }
+        self.datePicker = datePicker
+        return datePicker
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        chosenValueRow.titleLabel?.text = NSLocalizedString("Choose a time", comment: "Label for Publish time picker")
+        chosenValueRow.detailLabel?.text = datePicker.date.longStringWithTime()
+        let doneButton = UIBarButtonItem(title: NSLocalizedString("Done", comment: "Label for Done button"), style: .done, target: self, action: #selector(done))
+        navigationItem.setRightBarButton(doneButton, animated: false)
+    }
+
+    @IBAction func timePickerChanged(_ sender: Any) {
+        chosenValueRow.detailLabel?.text = datePicker.date.longStringWithTime()
+        coordinator?.setDate(datePicker.date)
+    }
+
+    @IBAction func done() {
+        coordinator?.updated(coordinator?.date)
+        navigationController?.dismiss(animated: true, completion: nil)
+    }
+}
+
+/// A base class used by the date picker classes
+/// Could be genericized further
+class DatePickerSheet: UIViewController {
+    var coordinator: DateCoordinator? = nil
+
+    weak var chosenValueRow: ChosenValueRow!
+    weak var pickerView: UIView!
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        WPStyleGuide.configureColors(view: view, tableView: nil)
+
+        let chosenValueRow = configureValueRow()
```

Comment:
We could assign directly to the instance variables instead of creating local variables.